include/core/memory.hpp

```diff
@@ -235,9 +272,7 @@ class Memory {
   /**
    * \brief This function gets called for every changed area in Memory
    */
-  std::function<void(const std::size_t, const std::size_t)> _callback = [](
-      const std::size_t, const std::size_t) {};
-  /**< Brief This function gets called for every changed area in Memory*/
+  CallbackFunction _callback = [](const std::size_t, const std::size_t) {};
```

Comment:
Do not define variable in the header! That is *very* un-C++-y! Use the constructor's initializer list